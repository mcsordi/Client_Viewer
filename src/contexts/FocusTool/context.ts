import { createContext, Dispatch, SetStateAction } from "react";
type TFocusTool = {
    focus: boolean
    setFocus: Dispatch<SetStateAction<boolean>>
}

export const focusTool = createContext({} as TFocusTool)
