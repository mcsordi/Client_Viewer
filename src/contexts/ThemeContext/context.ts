import { createContext } from "react";

interface IThemeProvider {
    light: boolean,
    changeTheme: () => void
}

export const themeContext = createContext({} as IThemeProvider)

