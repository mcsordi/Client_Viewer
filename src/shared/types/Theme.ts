export type TTheme = {
    theme?: boolean;
};
