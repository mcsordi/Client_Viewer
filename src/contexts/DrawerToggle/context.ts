import { createContext } from 'react';
import { TNavDrawer } from '../../shared/types/NavDrawer';


type TToggle = {
    show: Boolean
    toggleState: () => void
    drawerOptions: TNavDrawer[]
    handleDrawerOptions: (navDrawer: TNavDrawer[]) => void
}

export const toggle = createContext({} as TToggle)
