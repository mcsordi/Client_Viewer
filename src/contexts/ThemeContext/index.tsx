import React, { useCallback, useState } from 'react';
import { themeContext } from './context.ts';
import { Children } from '../../shared/types/Children.ts';

export const ThemeContext: React.FC<Children> = ({ children }) => {
  const [light, setLight] = useState<boolean>(true);
  const changeTheme = useCallback(() => {
    setLight((old) => !old);
  }, []);
  return (
    <themeContext.Provider value={{ light, changeTheme }}>
      {children}
    </themeContext.Provider>
  );
};
