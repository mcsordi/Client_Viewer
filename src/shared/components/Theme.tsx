import React, { useContext } from 'react';
import { themeContext } from '../../contexts/ThemeContext/context.ts';
import { MdLightMode } from 'react-icons/md';
import { MdDarkMode } from 'react-icons/md';

type ITheme = {
  theme?: boolean;
};

export const Theme: React.FC<ITheme> = ({ theme }) => {
  const { light, changeTheme } = useContext(themeContext);

  return (
    <button
      className={` ${light || (theme && `bg-slate-700`)} rounded-md absolute right-0 border p-2 bg-white`}
      onClick={() => changeTheme()}
    >
      {light || theme ? (
        <MdLightMode className="text-yellow-500 text-2xl" />
      ) : (
        <MdDarkMode className="text-2xl text-gray-700" />
      )}
    </button>
  );
};
