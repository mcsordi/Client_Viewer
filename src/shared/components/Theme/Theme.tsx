import React, { useContext } from 'react';
import { themeContext } from '../../../contexts/ThemeContext/context.ts';
import { MdLightMode } from 'react-icons/md';
import { MdDarkMode } from 'react-icons/md';
import { TTheme } from '../../types/Theme.ts';

export const Theme: React.FC<TTheme> = ({ theme }) => {
  const { light, changeTheme } = useContext(themeContext);

  return (
    <button
      className={` ${light || (theme && `bg-slate-700`)} rounded-md absolute right-0 border p-2 bg-white`}
      onClick={() => changeTheme()}
    >
      {light || theme ? (
        <MdLightMode className="text-yellow-500 text-2xl" />
      ) : (
        <MdDarkMode className="text-2xl text-gray-700" />
      )}
    </button>
  );
};
