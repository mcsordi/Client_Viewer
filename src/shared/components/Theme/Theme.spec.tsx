import { render, screen } from '@testing-library/react';
import { Theme } from './Theme';
import { themeContext } from '../../../contexts/ThemeContext/context';

describe('<Theme/>', () => {
  it('should render a theme button', () => {
    const changeTheme = () => {};
    render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <Theme />
      </themeContext.Provider>,
    );
    const textButton = screen.getByText('Alterar Tema');
    expect(textButton.parentElement).toHaveClass('bg-white absolute bottom-0');
  });
  it('should render a button with slate background', () => {
    const changeTheme = () => {};
    const { container } = render(
      <themeContext.Provider value={{ light: true, changeTheme }}>
        <Theme />
      </themeContext.Provider>,
    );
    expect(container.firstChild).toHaveClass('bg-slate-700');
    expect(container.firstChild?.firstChild).toHaveClass('text-yellow-500');
  });
  it('should render a button with white background', () => {
    const changeTheme = () => {};
    const { container } = render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <Theme />
      </themeContext.Provider>,
    );
    expect(container.firstChild).not.toHaveClass('bg-slate-700');
    expect(container.firstChild?.firstChild).toHaveClass('text-gray-700');
  });
  it('should match a snapshot', () => {
    const changeTheme = () => {};
    const { container } = render(
      <themeContext.Provider value={{ light: false, changeTheme }}>
        <Theme />
      </themeContext.Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
