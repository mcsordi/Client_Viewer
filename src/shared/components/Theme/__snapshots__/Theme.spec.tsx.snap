// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Theme/> should match a snapshot 1`] = `
<button
  class=" false rounded-md h-11 border p-2 bg-white"
>
  <svg
    class="text-2xl text-gray-700"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    stroke-width="0"
    viewBox="0 0 24 24"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M12 3a9 9 0 1 0 9 9c0-.46-.04-.92-.1-1.36a5.389 5.389 0 0 1-4.4 2.26 5.403 5.403 0 0 1-3.14-9.8c-.44-.06-.9-.1-1.36-.1z"
    />
  </svg>
</button>
`;
